src/main/java/hudson/plugins/robot/model/RobotSuiteResult.java

```diff
@@ -192,8 +197,11 @@ public void setEndTime(String endTime){
 
 	@Override
 	public long getDuration() {
-		if (StringUtils.isEmpty(this.startTime) || StringUtils.isEmpty(this.endTime))
-				return duration;
+		if (StringUtils.isNotEmpty(this.elapsedTime)) {
```

Comment:
Doesn't this now always choose `elapsedTime` as that is always present in all scenarios? In other words, should we remove the `else if` branch, or flip the logic so by default it would still check duration and then fall back to elapsedTime -- what do you think?

Reply:
By default that's "null", so normally it just goes pass that. Normally I think the whole if-else is bypassed and moved to calculation.